src/charm.py

```diff
@@ -242,6 +232,81 @@ def restart_mongos_service(self) -> None:
         self.stop_mongos_service()
         self.start_mongos_service()
 
+    def share_uri(self) -> None:
+        """Future PR - generate URI and give it to related app"""
+        # TODO future PR - generate different URI for data-integrator as that charm will not
+        # communicate to mongos via the Unix Domain Socket.
+
+    def set_user_role(self, roles: List[str]):
+        """Updates the roles for the mongos user."""
+        roles_str = roles.join(",")
+        self.app_peer_data[USER_ROLES_TAG] = roles_str
+
+        if len(self.model.relations[Config.Relations.CLUSTER_RELATIONS_NAME]) == 0:
+            return
+
+        # a mongos shard can only be related to one config server
+        config_server_rel = self.model.relations[
+            Config.Relations.CLUSTER_RELATIONS_NAME
+        ][0]
+        self.cluster.update_relation_data(
+            config_server_rel.id, {USER_ROLES_TAG: roles_str}
+        )
```

Comment:
I'd keep this one as a `self.cluster.database_requires.update_relation_data()`

Reply:
gotcha!